api/src/main/java/io/lumeer/api/model/UserNotification.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.lumeer.api.model;
+
+import io.lumeer.api.adapter.ZonedDateTimeAdapter;
+import io.lumeer.engine.api.data.DataDocument;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+
+/**
+ * @author <a href="mailto:marvenec@gmail.com">Martin Večeřa</a>
+ */
+public class UserNotification {
+
+   public static final String USER_ID = "userId";
+   public static final String CREATED_AT = "createdAt";
+   public static final String FIRST_READ_AT = "firstReadAt";
+   public static final String READ = "read";
+   public static final String DATA = "data";
+   public static final String TYPE = "type";
+
+   private String id;
+
+   private String userId;
+
+   @XmlJavaTypeAdapter(ZonedDateTimeAdapter.class)
+   private ZonedDateTime createdAt;
+
+   private boolean read = false;
+
+   @XmlJavaTypeAdapter(ZonedDateTimeAdapter.class)
+   private ZonedDateTime firstReadAt;
+
+   private NotificationType type;
+
+   private DataDocument data;
+
+   public UserNotification() {
+   }
+
+   public UserNotification(final String userId, final ZonedDateTime createdAt, final boolean read, final ZonedDateTime firstReadAt, final NotificationType type, final DataDocument data) {
```

Comment:
I think you should implement constructor with @JsonCreator annotation and @JsonProperty annonations for properties in order to work properly through Rest

Reply:
I forgot!